home/navigation/Navigation.tsx

```diff
@@ -229,7 +230,11 @@ export default (props: { theme: ColorTheme }) => {
       }
 
       if (url.startsWith('expo-home://qr-scanner')) {
-        nav.navigate('QRCode');
+        if (await requestCameraPermissionsAsync()) {
+          nav.navigate('QRCode');
+        } else {
+          alert('In order to use the QR Code scanner you need to provide camera permissions');
```

Comment:
ideally we avoid `alert` because the title is just "Alert" in that case:
<img width="314" alt="Screen Shot 2021-06-28 at 10 38 02 AM" src="https://user-images.githubusercontent.com/90494/123679925-eba32480-d7fc-11eb-895c-79fc18606dcb.png">

the [Alert](https://reactnative.dev/docs/alert) API gives us more control over the title. we may also want to tell people how they can go about enabling this permission now that they are at the point where they are seeing this dialog. we could possibly link to the app settings - you can do this with the react-native Linking api: `Linking.openSettings()`